terraform/versions.tf

```
@@ -5,7 +5,7 @@ terraform {
   required_providers {
     juju = {
       source  = "juju/juju"
-      version = ">= 0.22.0"
+      version = "~> 1.0.0"
```

```suggestion
      version = "~> 1.0"
```

Maybe change it to `~> 1.0`, since I feel `~> 1.0.0` is too narrow.